### DISCUSSION TITLE
SMTP Connection timeout

### DISCUSSION THREAD

**Discussion by maurxce** [outline/outline#7256] [Self Hosting] [answered]
Hello!

I just set up an outline v0.77.3 container and I tried inviting a different email of mine to test my SMTP settings, but I'm always getting a timeout:

```
{"error":"Connection timeout","level":"error","message":"Error processing task in EmailTask","stack":"Error: Connection timeout\n    at SMTPConnection._formatError (/opt/outline/node_modules/nodemailer/lib/smtp-connection/index.js:790:19)\n    at SMTPConnection._onError (/opt/outline/node_modules/nodemailer/lib/smtp-connection/index.js:776:20)\n    at Timeout.<anonymous> (/opt/outline/node_modules/nodemailer/lib/smtp-connection/index.js:235:22)\n    at listOnTimeout (node:internal/timers:573:17)\n    at process.processTimers (node:internal/timers:514:7)"}
```

**These are my SMTP settings:**
```env
SMTP_HOST=smtp.gmail.com
SMTP_PORT=465
SMTP_USERNAME=someemail@workspace.com
SMTP_PASSWORD=**********
SMTP_FROM_EMAIL=$SMTP_USERNAME
SMTP_REPLY_EMAIL=
SMTP_TLS_CIPHERS=SSLv3
SMTP_SECURE=true
```

**Some more info:**
- I tried `SMTP_TLS_CIPHERS=TLSv1.2`, but it didn't work
- The email address I'm using is a Google Workspace address
- The email I'm trying to invite is a regular gmail account
- I'm running outline behind nginx proxy manager

**Outline (please complete the following information):**
- Install: self hosted
- Version: 0.77.3
- ID: 25c1700a09e0

**Comment by tommoor** [comment]
Hi, if the SMTP connection is timing out then Outline cannot reach your server – this is not an issue with Outline itself, but with your configuration or networking setup. Maybe take a look at DNS resolution, it's the cause of lots of issues.

For example you can connect to the container and try SSHing to the smtp server directly to see if that works.

**Reply by maurxce** [reply]
Thanks for the suggestion!
I can ping `smtp.gmail.com` from my *server* just fine, I wasn't able to test it within the container though, as none of the following commands worked, any ideas on what I could try?:
- ping
- curl
- ssh

**Reply by tommoor** [reply]
You can install any package with apt-get, not much is installed by default to keep the distributed container tiny

**Reply by maurxce** [reply]
For that I need sudo rights, I can't find the root password anywhere
```
nodejs@784d9724cb56:/opt/outline$ apt install
E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
```

**Reply by tommoor** [reply]
Ah, then `wget` is installed – probably the only option

**Reply by maurxce** [reply]
`bash: wget: command not found` doesn't look too good right now 😬

**Comment by maurxce** [comment, marked as answer]
The issue has been solved by using this:
```env
SMTP_HOST=smtp.gmail.com
SMTP_PORT=587
SMTP_USERNAME=example@gmail.com
SMTP_PASSWORD=<app password>
SMTP_FROM_EMAIL=Outline <example@gmail.com>
SMTP_TLS_CIPHERS=TLSv1.2
SMTP_SECURE=false
```

**Reply by radueduard97** [reply]
Thanks a lot! Mine was acting out weird as well with Gmail
